compiler/z/codegen/OMRRegisterDependency.cpp

```diff
@@ -1051,7 +1051,9 @@ TR_S390RegisterDependencyGroup::assignRegisters(TR::Instruction   *currentInstru
 
       dependentRegNum = _dependencies[i].getRealRegister();
 
-      if (dependentRegNum != TR::RealRegister::SpilledReg)
+      if (dependentRegNum != TR::RealRegister::SpilledReg &&
+          dependentRegNum != TR::RealRegister::KillVolHighRegs &&
+          dependentRegNum != TR::RealRegister::NoReg)
```

Comment:
Can you explain this change?

Reply:
Around linkages we add a register dependency `KillVolHighRegs` only on 32 bit Java is 64 bit registers are allowed. After killing the high regs we change this register dependency to a NoReg. When these register dependencies get here we weren't ignoring them and counted them as  HPRs resulting in an incorrect count for HPRs.

https://github.com/eclipse/omr/blob/b9f595237b4f73dc0008bbf748c394e789fce17e/compiler/z/codegen/OMRLinkage.cpp#L2589-L2594

https://github.com/eclipse/omr/blob/b9f595237b4f73dc0008bbf748c394e789fce17e/compiler/z/codegen/OMRRegisterDependency.cpp#L1191-L1197